stdlib/Profile/test/allocs.jl

```diff
@@ -143,24 +147,29 @@ end
     @test length([a for a in prof.allocs if a.type == String]) >= 1
 end
 
+# FIXME: We currently do not call the allocation profiler
+# in the fastpath allocation for MMTk in codegen. Therefore, this
+# test has been disabled for MMTk.
 @testset "alloc profiler catches allocs from codegen" begin
-    @eval begin
-        struct MyType x::Int; y::Int end
-        Base.:(+)(n::Number, x::MyType) = n + x.x + x.y
-        foo(a, x) = a[1] + x
-        wrapper(a) = foo(a, MyType(0,1))
-    end
-    a = Any[1,2,3]
-    # warmup
-    wrapper(a)
+    if Base.USING_STOCK_GC
```

Comment:
Seems like you forgot the `@static` before the `if`.

Also, style NIT: probably better to guard the whole `testset`, as opposed to guarding its body?

E.g.

```Julia
@static if Base.USING_STOCK_GC
@testset "alloc profiler catches allocs from codegen" begin
...
end
end
```